### PR TITLE
[Feature] detrend R-R intervals

### PR DESCRIPTION
**Pull request by danibene**
# Description

This PR aims to add the ability to detrend R-R intervals as a preprocessing step. See also https://github.com/neuropsychology/NeuroKit/issues/671#issuecomment-1192127562

# Proposed Changes

I changed the `intervals_preprocess()` function so that it optionally calls `signal_detrend()` by passing `detrend_method`.

# Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#structure-and-code) file.
- [x] My PR is targetted at the **dev branch** (and not towards the master branch).
- [x] I ran the [CODE CHECKS](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#run-code-checks) on the files I added or modified and fixed the errors.
- [ ] I have added the newly added features to **News.rst** (if applicable)

### CODE REVIEW

File: neurokit2/hrv/intervals_preprocess.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    **kwargs
+        Keyword arguments to be passed to :func:`.signal_interpolate`.
+
+    Returns
+    -------
+    intervals : array
+        Preprocessed intervals, in milliseconds.
+    intervals_time : array
+        Preprocessed timestamps corresponding to intervals, in seconds.
+
```
Comment (DominiqueMakowski) [inline neurokit2/hrv/intervals_preprocess.py:42]: Could we add some examples here?

**Comment by DominiqueMakowski** [comment]
Awesome, last two little things; should we rename `intervals_preprocess()` to `intervals_process()` to maintain consistency with the other `*_process()` functions 

Also, we can add th new exported function to the [documentation](https://raw.githubusercontent.com/neuropsychology/NeuroKit/master/docs/functions/hrv.rst)

**Comment by DominiqueMakowski** [comment]
All good for me, feel free to merge!

**Comment by danibene** [comment]
> All good for me, feel free to merge!

Awesome, is it fine for me to merge https://github.com/neuropsychology/NeuroKit/pull/710 first? Since this is based off of that one

**Comment by DominiqueMakowski** [comment]
Yup!

**Pull request merged**